### ISSUE TITLE
Library cannot support more than one sensor

### ISSUE THREAD

**Issue by rah2501** [hcrest/sh2#9]
I'm using the [SparkFun VR IMU BNO08X Arduino Library](https://github.com/sparkfun/SparkFun_BNO08x_Arduino_Library) which imports this sh2 library.  Unfortunately, the API in the BNO08x library cannot support more than one sensor as it uses [global variables](https://github.com/sparkfun/SparkFun_BNO08x_Arduino_Library/blob/a78e3b442a7d33f109d2ca444e6febb54d44ec7a/src/SparkFun_BNO08x_Arduino_Library.cpp#L45) for sensor-specific data.  I've been updating the library to enable its use with more than one sensor.

Unfortunately, I've come across a similar issue with your sh2 library.  The BNO08x library calls [sh2_service()](https://github.com/ceva-dsp/sh2/blob/b514b1e2586ddc195e553dac89fc94c637b25298/sh2.c#L1929) which uses [another global variable named _sh2](https://github.com/ceva-dsp/sh2/blob/b514b1e2586ddc195e553dac89fc94c637b25298/sh2.c#L429) for sensor-specific state.

**Comment by rah2501** [comment]
Can I ask if there is a specific reason you used a global variable to store sensor state?  Are you open to changing the API so that the library can be used with more than one sensor?

**Comment by rah2501** [comment]
I've opened #10 as an example of how the API might be changed to support more than one sensor.  What are your thoughts?

**Comment by rah2501** [comment]
Hello? @kay-ceva ?

**Comment by WolfgangWindholtz** [comment]
@rah2501 Have you figured this issue this issue out. We are trying figure out the same issue for a project.

**Comment by rah2501** [comment]
> @rah2501 Have you figured this issue this issue out.

No.  I don't want to waste time making changes to the library that might not be accepted by upstream.  I'm still waiting on the maintainers (Ceva) to respond.  It looks like they might not be interested in collaborative development.

@dwheeler-ceva ?

**Comment by iLoveAndyBaker** [comment]
If the upstream people are not going to make the changes, through no bad faith on their part of course, then why doesn't the community do it instead? It seems just as likely that the upstream folks won't make ANY new changes anyway.

**Comment by rah2501** [comment]
> why doesn't the community do it instead?

Because the community doesn't have write privileges to the repository.

**Comment by iLoveAndyBaker** [comment]
Rather I should say, leave the upstream version well enough alone, and pack a "fixed" version in the Sparkfun library or whoever happens to be using it

**Comment by dwheeler-ceva** [comment]
The code in this repo was originally intended as an example implementation, not necessarily as a ready-to-use driver.  We chose to support just one device because it was the simplest, most universal case.  (For similar reasons, there's no OS integration here.)  

I guess what I need to say here is you should consider forking this repo and working together to implement the changes you need.
